codes/ruby/chapter_backtracking/preorder_traversal_iii_template.rb

```diff
@@ -0,0 +1,69 @@
+=begin
+File: preorder_traversal_iii_template.rb
+Created Time: 2024-04-07
+Author: hoangtien2k3 (hoangtien2k3qx1@163.com)
+=end
+
+require_relative 'utils'
+
+# 判断当前状态是否为解
+def is_solution(state)
+  !state.empty? && state[-1].val == 7
+end
+
+# 记录解
+def record_solution(state, res)
+  res << state.dup
+end
+
+# 判断在当前状态下，该选择是否合法
+def is_valid(state, choice)
+  !choice.nil? && choice.val != 3
```

Comment:
The code is not consistent with other codes

```suggestion
  choice if !choice.nil? && choice.val != 3
```